libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/StepManeuverType.kt

```diff
@@ -0,0 +1,39 @@
+package com.mapbox.navigation.base.route.model
+
+import androidx.annotation.StringDef
+
+const val TURN = "turn"
+const val NEW_NAME = "new name"
+const val DEPART = "depart"
+const val ARRIVE = "arrive"
+const val MERGE = "merge"
+const val ON_RAMP = "on ramp"
+const val OFF_RAMP = "off ramp"
+const val FORK = "fork"
+const val END_OF_ROAD = "end of road"
+const val CONTINUE = "continue"
+const val ROUNDABOUT = "roundabout"
+const val ROTARY = "rotary"
+const val ROUNDABOUT_TURN = "roundabout turn"
+const val NOTIFICATION = "notification"
+const val EXIT_ROUNDABOUT = "exit roundabout"
+const val EXIT_ROTARY = "exit rotary"
+
+@Retention(AnnotationRetention.SOURCE)
+@StringDef(TURN,
```

Comment:
What about using [`sealed class`es](https://kotlinlang.org/docs/reference/sealed-classes.html#sealed-classes) instead of `@StringDef`s?

Reply:
Same as above. Converting it to `sealed` class will associate state with it. I don't think we should store `state`.